### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by david-waltermire**
# Committer Notes

Code cleanup and addressing PMD and Spotbug warnings.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/metaschema-framework/metaschema-java/blob/main/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/metaschema-framework/metaschema-java/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [ ] Have you updated all website and readme documentation affected by the changes you made?

### CODE REVIEW

File: cli-processor/src/main/java/gov/nist/secauto/metaschema/cli/processor/command/CommandExecutionException.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   * @param message
+   *          the exception message
+   */
+  public CommandExecutionException(@NonNull ExitCode code, String message) {
+    super(message);
+    this.exitCode = code;
+  }
+
+  /**
+   * Constructs a new exception with the no message and provided the {@code code}
```
Comment (aj-stein-gsa) [inline cli-processor/src/main/java/gov/nist/secauto/metaschema/cli/processor/command/CommandExecutionException.java:51]: Very minor, non-blocking: grammar in docstring.

```suggestion
   * Constructs a new exception with no message and provided the {@code code}
```

File: core/src/main/java/gov/nist/secauto/metaschema/core/util/DeleteOnShutdown.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                @Override
+                public FileVisitResult visitFile(
+                    Path file, BasicFileAttributes attrs)
+                    throws IOException {
+                  Files.delete(file);
+                  return FileVisitResult.CONTINUE;
+                }
+              });
+        } catch (IOException ex) {
+          // this is a best effort
```
Comment (aj-stein-gsa) [inline core/src/main/java/gov/nist/secauto/metaschema/core/util/DeleteOnShutdown.java:58]: Non-blocking: perhaps warn that DeleteOnShutdown behavior is best effort and cleanup was not complete successful, just to be on the safe side? Could help us debug later if it comes to that.

File: databind/src/main/java/gov/nist/secauto/metaschema/databind/io/xml/MetaschemaXmlReader.java [hunk 3]
```diff
@@ -146,8 +146,9 @@ public <CLASS> CLASS read(@NonNull IBoundDefinitionModelComplex definition) thro
       ItemReadHandler handler = new ItemReadHandler(ObjectUtils.notNull(event.asStartElement()));
       Object value = definition.readItem(null, handler);
       if (value == null) {
+        event = reader.peek();
         throw new IOException(String.format("Unable to read data%s",
```
Comment (aj-stein-gsa) [inline databind/src/main/java/gov/nist/secauto/metaschema/databind/io/xml/MetaschemaXmlReader.java:150]: Minor, non-blocking: not material to the commit, but do we want to space between the word data and formatting string?

```suggestion
        throw new IOException(String.format("Unable to read data %s",
```

File: metaschema-cli/src/main/java/gov/nist/secauto/metaschema/cli/commands/ValidateModuleCommand.java [hunk 4]
```diff
@@ -90,16 +93,17 @@ protected ISchemaValidationProvider getSchemaValidationProvider(
...
       try (InputStream is = this.getClass().getResourceAsStream("/schema/xml/metaschema-model_schema.xsd")) {
-        List<Source> retval = new LinkedList<>();
-        retval.add(new StreamSource(
+        List<Source> sources = new LinkedList<>();
+        sources.add(new StreamSource(
             ObjectUtils.requireNonNull(is,
                 "Unable to load '/schema/xml/metaschema.xsd' on the classpath")));
-        return new XmlSchemaContentValidator(retval);
+        return new XmlSchemaContentValidator(sources);
       }
```
Comment (aj-stein-gsa) [inline metaschema-cli/src/main/java/gov/nist/secauto/metaschema/cli/commands/ValidateModuleCommand.java:107]: Minor, non-blocking: this is good, but in the future can we consider https://github.com/metaschema-framework/metaschema-java/issues/231 for a later enhancement outside the scope of this review?

**Review by aj-stein-gsa — DISMISSED** [review]

File: cli-processor/src/main/java/gov/nist/secauto/metaschema/cli/processor/command/CommandExecutionException.java (hunk 1 above)
Comment (david-waltermire) [reply cli-processor/src/main/java/gov/nist/secauto/metaschema/cli/processor/command/CommandExecutionException.java:51]:
```suggestion
   * Constructs a new exception with no message and the provided {@code code}
```

File: core/src/main/java/gov/nist/secauto/metaschema/core/util/DeleteOnShutdown.java (hunk 2 above)
Comment (david-waltermire) [reply core/src/main/java/gov/nist/secauto/metaschema/core/util/DeleteOnShutdown.java:58]: I don't think this is possible due to how the shutdown works. Logging might be shutdown already.

File: databind/src/main/java/gov/nist/secauto/metaschema/databind/io/xml/MetaschemaXmlReader.java (hunk 3 above)
Comment (david-waltermire) [reply databind/src/main/java/gov/nist/secauto/metaschema/databind/io/xml/MetaschemaXmlReader.java:150]: The message that gets inserted has the space.
```suggestion
        throw new IOException(String.format("Unable to read data.%s",
```

File: core/src/main/java/gov/nist/secauto/metaschema/core/util/DeleteOnShutdown.java (hunk 2 above)
Comment (aj-stein-gsa) [reply core/src/main/java/gov/nist/secauto/metaschema/core/util/DeleteOnShutdown.java:58]: Understandable, I agree with resolving the comment and moving along.

**Review by aj-stein-gsa — APPROVED** [review]

**Pull request merged**